tests/e2e/system/test_mcg_recovery.py

```diff
@@ -0,0 +1,219 @@
+            mcg_obj=mcg_obj_session,
+        )
+
+        random_object_round_trip_verification(
+            io_pod=awscli_pod_session,
+            bucket_name=second_bidi_bucket_name,
+            upload_dir=bidi_uploaded_objs_dir_2,
+            download_dir=bidi_downloaded_objs_dir_2,
+            amount=object_amount,
+            pattern="SecondBidi-",
+            wait_for_replication=True,
+            second_bucket_name=first_bidi_bucket_name,
+            mcg_obj=mcg_obj_session,
+        )
+
+        # Create a cache bucket
+        cache_bucketclass = {
+            "interface": "OC",
+            "namespace_policy_dict": {
+                "type": "Cache",
+                "ttl": 3600000,
+                "namespacestore_dict": {
+                    "aws": [(1, "eu-central-1")],
+                },
+            },
+            "placement_policy": {
+                "tiers": [{"backingStores": [constants.DEFAULT_NOOBAA_BACKINGSTORE]}]
+            },
+        }
+        cache_bucket = bucket_factory(bucketclass=cache_bucketclass)[0]
+
+        cache_uploaded_objs_dir = uploaded_objects_dir + "/cache"
+        cache_uploaded_objs_dir_2 = uploaded_objects_dir + "/cache_2"
+        cache_downloaded_objs_dir = downloaded_obejcts_dir + "/cache"
+        underlying_bucket_name = cache_bucket.bucketclass.namespacestores[0].uls_name
+
+        # Upload a random object to the bucket
+        objs_written_to_cache_bucket = write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            cache_bucket.name,
+            cache_uploaded_objs_dir,
+            mcg_obj=mcg_obj_session,
+        )
+        wait_for_cache(mcg_obj_session, cache_bucket.name, objs_written_to_cache_bucket)
+        # Write a random, larger object directly to the underlying storage of the bucket
+        write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            underlying_bucket_name,
+            cache_uploaded_objs_dir_2,
+            s3_creds=cld_mgr.aws_client.nss_creds,
+        )
+        # Download the object from the cache bucket
+        sync_object_directory(
+            awscli_pod_session,
+            f"s3://{cache_bucket.name}",
+            cache_downloaded_objs_dir,
+            mcg_obj_session,
+        )
+        # Make sure the cached object was returned, and not the one that was written to the underlying storage
+        assert compare_directory(
+            awscli_pod_session,
+            cache_uploaded_objs_dir,
+            cache_downloaded_objs_dir,
+            amount=1,
+        ), "The uploaded and downloaded cached objects have different checksums"
+        assert (
+            compare_directory(
+                awscli_pod_session,
+                cache_uploaded_objs_dir_2,
+                cache_downloaded_objs_dir,
+                amount=1,
+            )
+            is False
+        ), "The cached object was replaced by the new one before the TTL has expired"
+
```

Comment:
@OdedViner these preconditions/setup would be common to many of our system test cases. I would suggest moving to the tests/e2e/system/conftest.py file so that we can use this as base_setup_fixture for all our system-test cases. wdyt?

Reply:
If @OdedViner and @hnallurv will endorse the offer, I will implement it next week

Reply:
+1 to Akarha's comment. Ben, please go ahead.

Reply:
Will implement and update the PR soon

Reply:
The offer was implemented.
The `setup_mcg_system` fixture now returns a function that can be used for any test. It returns the function in order to enable parametrization of bucket and object amounts.
Then, the return value of `setup_mcg_system` should be sent to `verify_mcg_system_recovery` at the end of the test; I did not implement it as a teardown fixture because these checks are part of the test, not the teardown. 

The MCG parts were verified locally and work as expected.